timesketch/api/v1/resources/graph.py

```diff
@@ -34,19 +39,90 @@ class GraphListResource(resources.ResourceMixin, Resource):
     """Resource to get all graphs."""
 
     @login_required
-    def get(self):
+    def get(self, sketch_id):
         """Handles GET request to the resource.
 
         Returns:
             List of graphs in JSON (instance of flask.wrappers.Response)
         """
-        graphs = manager.GraphManager.get_graphs()
-        graph_names = {name: graph_class().DISPLAY_NAME for name, graph_class in
-                       graphs}
-        return jsonify(graph_names)
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        graphs = sketch.graphs
+        return self.to_json(graphs)
+
+    def post(self, sketch_id):
+        """Handles POST request to the resource."""
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        form = request.json
+        name = form.get('name')
+        elements = form.get('elements')
+
+        graph = Graph(
+            user=current_user, sketch=sketch, name=str(name),
+            graph_elements=json.dumps(elements))
+        db_session.add(graph)
+        db_session.commit()
+
+        return self.to_json(graph)
 
 
 class GraphResource(resources.ResourceMixin, Resource):
+    """Resource to get all graphs."""
+
+    @login_required
+    def get(self, sketch_id, graph_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of graphs in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        graph = Graph.query.get(graph_id)
+
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        if not sketch.id == graph.sketch.id:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                'Graph does not belong to this sketch.')
+
+        response = self.to_json(graph).json
+        response['objects'][0]['elements'] = graph.graph_elements
+
+        return jsonify(response)
+
+
+class GraphCacheListResource(resources.ResourceMixin, Resource):
+    """Resource to get all graphs."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of graphs in JSON (instance of flask.wrappers.Response)
+        """
+        graph_plugins = manager.GraphManager.get_graphs()
+        graphs = []
+        for name, graph_class in graph_plugins:
+            graph_instance = graph_class()
```

Comment:
why do you need to instantiate the class to get access to these constants? You should be able to use the class instance here

Reply:
good point, fixed.